subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/cpu.h

```diff
@@ -16,12 +16,21 @@ static inline void cpu_sleep(void)
 #endif
 }
 
-static inline void cpu_dsb(void)
+static inline void cpu_dmb(void)
 {
-#if defined(CONFIG_CPU_CORTEX_M0) || defined(CONFIG_ARCH_POSIX)
-	/* No need of data synchronization barrier */
-#elif defined(CONFIG_CPU_CORTEX_M4) || defined(CONFIG_CPU_CORTEX_M33)
-	__DSB();
+#if defined(CONFIG_CPU_CORTEX_M0) || \
```

Comment:
Why not use `CONFIG_CPU_CORTEX_M`?

Reply:
> Why not use CONFIG_CPU_CORTEX_M?

It is just a choice to have explicit CPU check, to the architecture variants supported by controller, so that new ports are aware of this interface use.

Reply:
If the problem is the instruction re-ordering, i think we should try to solve it without asking the CPU to execute any instruction (DMB would eat 2 cycles, i think)

Reply:
> It is just a choice to have explicit CPU check, to the architecture variants supported by controller, so that new ports are aware of this interface use.

I guess it's better to hide it under SOC_NORDIC_NRF5_COMPATIBLE or similar.
Otherwise, CPU_CORTEX_M is cleaner, I think.